packages/contracts-bedrock/contracts/L2/L1Block.sol

```diff
@@ -64,10 +64,16 @@ contract L1Block {
             revert OnlyDepositor();
         }
 
-        number = _number;
-        timestamp = _timestamp;
-        basefee = _basefee;
-        hash = _hash;
-        sequenceNumber = _sequenceNumber;
+        bytes32 slot;
+        assembly {
+            // _number, _timestamp and _sequenceNumber are all uint64 and will
```

Comment:
If solidity does slot packing on layout & reads, can it do it on writes as well?

Reply:
Yup it can, it's slightly more expensive and hides the fact that they are in the same storage slot which I think is the benefit of doing it in yul